File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/UserChangeEvent.java
package work.socialhub.kslack.entity.event;

import com.github.seratch.jslack.api.model.User;

/**
 * The user_change event is sent to all connections for a workspace when a member updates their profile or data.
 * Clients can use this to update their local cache of members.
 * <p>
 * Use the users.profile.set method to update user profile data.
 * <p>
 * https://api.slack.com/events/user_change
 */
public class UserChangeEvent implements Event {

    public static final String TYPE_NAME = "user_change";

    private final String type = TYPE_NAME;
    private User user; // TODO: make sure the available attributes

    public String getType() {
        return this.type;
    }

    public User getUser() {
        return this.user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/UserChangeEvent.kt
package work.socialhub.kslack.entity.event

import work.socialhub.kslack.entity.user.User

/**
 * The user_change event is sent to all connections for a workspace when a member updates their profile or data.
 * Clients can use this to update their local cache of members.
 *
 *
 * Use the users.profile.set method to update user profile data.
 *
 *
 * https://api.slack.com/events/user_change
 */
class UserChangeEvent : Event {
    override val type: String = TYPE_NAME
    var user: User? = null // TODO: make sure the available attributes

    companion object {
        const val TYPE_NAME: String = "user_change"
    }
}